WickedEngine/wiHelper.cpp

```diff
@@ -112,6 +112,197 @@ namespace wi::helper
+
+			for (uint32_t i = 0; i < data_count; ++i)
+			{
+				XMFLOAT4 pixel = dataSrc[i];
+				float r = std::max(0.0f, std::min(pixel.x, 1.0f));
+				float g = std::max(0.0f, std::min(pixel.y, 1.0f));
+				float b = std::max(0.0f, std::min(pixel.z, 1.0f));
+				float a = std::max(0.0f, std::min(pixel.w, 1.0f));
+
+				uint32_t rgba8 = 0;
+				rgba8 |= (uint32_t)(r * 255.0f) << 0;
+				rgba8 |= (uint32_t)(g * 255.0f) << 8;
+				rgba8 |= (uint32_t)(b * 255.0f) << 16;
+				rgba8 |= (uint32_t)(a * 255.0f) << 24;
+
+				data32[i] = rgba8;
+			}
+			successfully_converted = true;
+		}
+		else if (desc.format == Format::R16G16B16A16_FLOAT)
+		{
+			// This will be converted first to rgba8 before saving to common format:
+			XMHALF4* dataSrc = (XMHALF4*)texturedata.data();
+			uint32_t* data32 = (uint32_t*)texturedata.data();
+
+			for (uint32_t i = 0; i < data_count; ++i)
+			{
+				XMHALF4 pixel = dataSrc[i];
+				float r = std::max(0.0f, std::min(XMConvertHalfToFloat(pixel.x), 1.0f));
+				float g = std::max(0.0f, std::min(XMConvertHalfToFloat(pixel.y), 1.0f));
+				float b = std::max(0.0f, std::min(XMConvertHalfToFloat(pixel.z), 1.0f));
+				float a = std::max(0.0f, std::min(XMConvertHalfToFloat(pixel.w), 1.0f));
+
+				uint32_t rgba8 = 0;
+				rgba8 |= (uint32_t)(r * 255.0f) << 0;
+				rgba8 |= (uint32_t)(g * 255.0f) << 8;
+				rgba8 |= (uint32_t)(b * 255.0f) << 16;
+				rgba8 |= (uint32_t)(a * 255.0f) << 24;
+
+				data32[i] = rgba8;
+			}
+			successfully_converted = true;
+		}
+		else if (desc.format == Format::R11G11B10_FLOAT)
+		{
+			// This will be converted first to rgba8 before saving to common format:
+			XMFLOAT3PK* dataSrc = (XMFLOAT3PK*)texturedata.data();
+			uint32_t* data32 = (uint32_t*)texturedata.data();
+
+			for (uint32_t i = 0; i < data_count; ++i)
+			{
+				XMFLOAT3PK pixel = dataSrc[i];
+				XMVECTOR V = XMLoadFloat3PK(&pixel);
+				XMFLOAT3 pixel3;
+				XMStoreFloat3(&pixel3, V);
+				float r = std::max(0.0f, std::min(pixel3.x, 1.0f));
+				float g = std::max(0.0f, std::min(pixel3.y, 1.0f));
+				float b = std::max(0.0f, std::min(pixel3.z, 1.0f));
+				float a = 1;
+
+				uint32_t rgba8 = 0;
+				rgba8 |= (uint32_t)(r * 255.0f) << 0;
+				rgba8 |= (uint32_t)(g * 255.0f) << 8;
+				rgba8 |= (uint32_t)(b * 255.0f) << 16;
+				rgba8 |= (uint32_t)(a * 255.0f) << 24;
+
+				data32[i] = rgba8;
+			}
+			successfully_converted = true;
+		}
+		if (rgba_to_bgra == true)
+		{
+			uint32_t* data32 = (uint32_t*)texturedata.data();
+			for (uint32_t i = 0; i < data_count; ++i)
+			{
+				uint32_t pixel = data32[i];
+				float b = ((pixel >> 0) & 255) / 255.0f;
+				float g = ((pixel >> 8) & 255) / 255.0f;
+				float r = ((pixel >> 16) & 255) / 255.0f;
+				float a = ((pixel >> 24) & 255) / 255.0f;
+
+				uint32_t rgba8 = 0;
+				rgba8 |= (uint32_t)(r * 255.0f) << 0;
+				rgba8 |= (uint32_t)(g * 255.0f) << 8;
+				rgba8 |= (uint32_t)(b * 255.0f) << 16;
+				rgba8 |= (uint32_t)(a * 255.0f) << 24;
+				data32[i] = rgba8;
+			}
```

Comment:
`successfully_converted = true;` seems to be missing from here

Reply:
Thanks for the immediate reply :)  Happy to expand contributions as needed.  Merging and syncing with the master version makes a lot of sense for MAX.  Just going through a few lines to sync our code with your new naming system and files. Naturally, I found the WICKEDENGINE_BACKWARDS_COMPATIBILITY_0_59 too late, so proceeding to do a full conversion, is probably for the best.  Will commit more once I am compiling again.